### ISSUE TITLE
Provide DOI numbers for containers

### ISSUE THREAD

**Issue by apeltzer** [singularityhub/sregistry#44]
Hi! 
an idea we discussed on our retreat recently was the possibility to have DOIs for containers within a certain registry. This would make it possible to state e.g. in a publication:

- We utilized software X in container Y (DOI: XXXX/YYYY/ZZZZ)

And thus be even consistently able to tell someone how you did a certain kind of analysis (when you also include parameters etc.). Would this be feasible at all or maybe feasible in Singularity hub (not sure whether this is the right place to ask for it?)

**Comment by vsoch** [comment]
This is a great idea, and I think perhaps would be possible (although not imminently because I'm a bit backed up with things to do!) 

For a proper DOI, it's a little bit more than a unique resource identifier I think - there are particular agencies that are authorized to grant them (e.g., read here https://www.doi.org/faq.html)

What I think would be reasonable to do is some kind of integration with a service that does grant DOIs. For example, http://developers.zenodo.org/ has an API. Minimally, we could point the user to generate a DOI for software (and have the container linked).

In terms of the number that you see in the url for the container, I wouldn't consider that a set in stone, forever kept identifier for the container - this is actually just an index for the database, and if singularity hub were re-created, it could be different. I think the link that we would want associated with the doi is for the specific container, for example, the "container" version of [https://singularity-hub.org/collections/vsoch/hello-world/](https://singularity-hub.org/collections/vsoch/hello-world/).

Right now, you could link to a particular container collection via a specific url, but it doesn't parse the tag or commit (well, the API endpoint does to pull a particular one, but not the web interface) so we would want to add that.

I'll put this as a requested feature for now - I think it's a huge plus + and I just need to collect some time to properly implement.

**Comment by sven1103** [comment]
Check this out: https://guides.github.com/activities/citable-code/

I am not sure if it is necessary to have it on shub, when you can already do it from GitHub?

What do you think is the advantage of using the i.e. Zenodo API from a registry?

**Comment by vsoch** [comment]
Yes, zenodo was the exact idea! We don't need to have Singularity Registry provide any sort of service for doing this, but rather just point the user to Zenodo. I pointed to the API because it would make it possible for us to have a button with a container that says something along the lines of "Generate a DOI" and then take the user to one of their views, perhaps with some of the information filled in / provided via the API. Zenodo does link to a Github repo, but it still would require upload of files. It probably (although I haven't checked) would be non trivial to upload huge containers (why singularity hub/registry would be needed to help).

TLDR, there is no critical advantage other than pointing the user to do it, and perhaps making it slightly easier. There are other integrations that might be useful too, although I've stayed away from them because they are very "I'm an academic researcher" specific, and I think Singularity containers are scoped well beyond that.

**Comment by vsoch** [comment]
This issue hasn't had comment for years, and functionality seems met with Zenodo (linked to the repository) so I'm going to close here. Please add further comment if needed!
